handler/transport.go

```diff
@@ -36,25 +38,36 @@ type transport struct {
 	cfg         config.Proxy
 	noAuthPaths []*policy.Assertion
 	// List to check for deprecated cipher suites
-	insecureCipherSuites []*tls.CipherSuite
+	insecureCipherSuites   []*tls.CipherSuite
+	latencyInstrumentation prometheus.Histogram
 }
 
 // Based on the following.
 // https://github.com/golang/oauth2/blob/bf48bf16ab8d622ce64ec6ce98d2c98f916b6303/transport.go
 func (t *transport) RoundTrip(r *http.Request) (*http.Response, error) {
+	var startTime time.Time
+
+	if t.latencyInstrumentation != nil {
+		defer func() {
+			endTime := time.Since(startTime)
+			t.latencyInstrumentation.Observe(float64(endTime.Seconds()))
```

Comment:
```suggestion
            // 1. skip metrics if request is not forwarded to origin
            // if (startTime == 0) {return}
			
            // 2. simplify

            // 3. use generic method
			t.metrics.Observe(metrics.LATENCY, nil, float64(time.Since(startTime).Seconds()))
```

Reply:
@WindzCUHK fixedin https://github.com/AthenZ/authorization-proxy/pull/30/commits/1a0bc8cef3624f605e59ddad85deec897964644b